File: CS/frmConvertStocks.cs
// VBConversions Note: VB project level imports
using System.Data;
using System.Collections.Generic;
using PDSA.Data;
using PDSA.WinForms.Forms;
using PDSA.Security;
using System.Diagnostics;
using PDSA.Common;
using PDSA.WinForms;
using PDSA.WinForms.Security;
using PDSA.PDSACryptography;
using DoorsPOS.Utility;
using Microsoft.VisualBasic;
using System.Collections;
using System;
using DoorsPOS.BusinessLayer;
using System.Windows.Forms;
using System.Drawing;
using System.Linq;
using DoorsPOS.DataLayer;
// End of VB project level imports

using PDSA.DataLayer.DataClasses;
using PDSA.Validation;
using DevExpress.XtraEditors;

namespace DoorsPOS.WinApp
{
	public partial class frmConvertStocks
	{
		public frmConvertStocks()
		{
			InitializeComponent();
		}
		private PDSA.DataLayer.DataClasses.PDSATransaction TranBo;
		stocksManager mgr8;
		int mOrderId = 0;
		PDSA.DataLayer.DataClasses.PDSATransaction Tran2;
		PDSA.DataLayer.DataClasses.PDSATransaction Tran3;
		public void frmConvertStocks_FormClosed(object sender, FormClosedEventArgs e)
		{
			this.Dispose();
		}
		
		public void frmConvertStocks_FormClosing(object sender, FormClosingEventArgs e)
		{
			if (btnSave.Enabled == true)
			{
				e.Cancel = true;
				MessageBox.Show("Save or Cancel the Conversion of Items First Before Closing the Form.");
			}
		}
		
		public void frmConvertStocks_Load(object sender, EventArgs e)
		{
			this.Cursor = Cursors.WaitCursor;
			
			
			this.Cursor = Cursors.Default;
		}
		
		public void btnItemSearch_Click(object sender, EventArgs e)
		{
			frmItemSearch frmitemserch = new frmItemSearch();
			frmitemserch.ShowDialog();
			txtItem.Text = string.Empty;
			txtStckId.Text = string.Empty;
			txtItem.Text = frmitemserch.vItem.ToString();
			txtStckId.Text = frmitemserch.vStckid.ToString();
			ceInner.Value = frmitemserch.vInnerQty;
			ceAvlbl.Value = frmitemserch.vAvlbl;
			
			if (txtStckId.Text == string.Empty)
			{
				return;
			}
			else
			{
				ItemsearchExecute();
			}
			btnItemsearch2.Focus();
		}
		
		private void ItemsearchExecute()
		{
			stocksCollection col2;
			int nStckid = 0;
			try
			{
				mgr8 = new stocksManager();
				mgr8.DataObject.WhereFilter = stocksData.WhereFilters.PrimaryKey;
				mgr8.DataObject.Entity.stckid = int.Parse(txtStckId.Text);
				col2 = mgr8.BuildCollection();
				if (mgr8.DataObject.RowsAffected > 0)
				{
					nStckid = mgr8.DataObject.Entity.stckid;
					//txtPrice.Text = FormatNumber(CStr(mgr8.DataObject.Entity.cost), 2)
					ceQty.Focus();
				}
				
			}
			catch (PDSA.Validation.PDSAValidationException ex)
			{
				MessageBox.Show(ex.Message);
			}
			catch (Exception ex)
			{
				MessageBox.Show(ex.ToString());
			}
		}
		
		public void btnClose_Click(object sender, EventArgs e)
		{
			this.Close();
		}
		
		public void btnSave_Click(object sender, EventArgs e)
		{
			if (cePcs.Value <= 1)
			{
				MessageBox.Show("Cannot convert quantity");
				return;
			}
			if (ceQty.Value > ceAvlbl.Value)
			{
				MessageBox.Show("Quantity to be converted must not be greater than the available stock.");
				return;
			}
			if (txtItem.Text == string.Empty)
			{
				MessageBox.Show("Please select an item to be converted.");
				txtItem.Focus();
				return;
			}
			
			if (ceQty.Value <= 0)
			{
				MessageBox.Show("Please enter the number of items to be converted.");
				ceQty.Focus();
				return;
			}
			
			if (deBoDate.Text == string.Empty)
			{
				MessageBox.Show("Please enter the date the item was converted.");
				deBoDate.Focus();
				return;
			}
			
			if (ceInner.Value < 2)
			{
				MessageBox.Show("The Item Innerbox Qty is not valid and should be at least 2.");
				ceInner.Focus();
				return;
			}
			
			
			TranBo = new PDSA.DataLayer.DataClasses.PDSATransaction(); // did not change but the transaction is about converting items to pieces
			TranBo.AfterSubmit += TranBo_AfterSubmit;
			TranBo.BeforeSubmit += Tran_BeforeSubmit;
			Tran2 = new PDSA.DataLayer.DataClasses.PDSATransaction();
			Tran3 = new PDSA.DataLayer.DataClasses.PDSATransaction();
			convertstckManager mgrbo = new convertstckManager(); // mgrbo is just a variable but it is pertaining to the convertstck table
			
			try
			{
				mgrbo.DataObject.TransactionType = PDSATransactionType.Insert;
				mgrbo.DataObject.Entity.innerqty = (int) ceInner.Value;
				mgrbo.DataObject.Entity.condate = DateTime.Parse(deBoDate.Text);
				mgrbo.DataObject.Entity.fromstckid = int.Parse(txtStckId.Text);
				mgrbo.DataObject.Entity.tostckid = int.Parse(txtstckid2.Text);
				mgrbo.DataObject.Entity.fromqty = (int) ceQty.Value;
				mgrbo.DataObject.Entity.InsertedBy = System.Convert.ToString(PDSAAppConfig.CurrentLoginID);
				mgrbo.DataObject.Entity.ModifiedBy = System.Convert.ToString(PDSAAppConfig.CurrentLoginID);
				mgrbo.DataObject.Entity.InsertedOn = DateTime.Now;
				mgrbo.DataObject.Entity.toqty = (int) cePcs.Value;
				TranBo.Add(mgrbo.DataObject);
				TranBo.Execute();
				
				spConvertFirstManager mgrspBo = new spConvertFirstManager();
				mgrspBo.Entity.StckIid = int.Parse(txtStckId.Text);
				mgrspBo.Entity.Convertid = mOrderId;
				Tran2.Add(mgrspBo.DataObject);
				//TranBo.Add(mgrspBo.DataObject)
				
				spConvertsecondManager mgrspconsec = new spConvertsecondManager();
				mgrspconsec.Entity.StckIid = int.Parse(txtstckid2.Text);
				mgrspconsec.Entity.Convertid = mOrderId;
				Tran2.Add(mgrspconsec.DataObject);
				//TranBo.Add(mgrspconsec.DataObject)
				
				Tran2.Execute();
				//TranBo.Execute()
				DisableControls();
				MessageBox.Show("Conversion Entry Successfully Saved", "Saved");
				InitControls();
			}
			catch (PDSA.Validation.PDSAValidationException ex)
			{
				MessageBox.Show(ex.Message);
				return;
			}
			catch (Exception ex)
			{
				MessageBox.Show(ex.ToString());
				return;
			}
			
		}
		private void TranBo_AfterSubmit(object sender, PDSA.DataLayer.DataClasses.PDSATransactionEventArgs e)
		{
			// Get the SQL Identity generated and store into local variable
			if (e.ClassName == "convertstckData")
			{
				mOrderId = ((convertstck) e.DataClassTable.EntityObject).convertid;
			}
		}
		
		private void Tran_BeforeSubmit(object sender, PDSA.DataLayer.DataClasses.PDSATransactionEventArgs e)
		{
			// Put local order id into line item objects prior to submitting INSERT statement
			if (e.ClassName == "convertstckData")
			{
				((convertstck) e.DataClassTable.EntityObject).convertid = mOrderId;
			}
		}
		
		private void EnableControls()
		{
			btnSave.Enabled = true;
			btnCancel.Enabled = true;
			GroupControl1.Enabled = true;
			btnClose.Enabled = false;
			btnNew.Enabled = false;
			deBoDate.Focus();
		}
		
		private void DisableControls()
		{
			btnSave.Enabled = false;
			btnCancel.Enabled = false;
			GroupControl1.Enabled = false;
			btnClose.Enabled = true;
			btnNew.Enabled = true;
			btnNew.Focus();
		}
		
		private void InitControls()
		{
			deBoDate.EditValue = DateAndTime.Today;
			txtItem.Text = "";
			txtitem2.Text = "";
			txtStckId.Text = "";
			txtstckid2.Text = "";
			ceQty.Value = 0;
			cePcs.Value = 0;
			ceAvlbl.Value = 0;
			
		}
		
		public void btnCancel_Click(object sender, EventArgs e)
		{
			DisableControls();
			InitControls();
		}
		
		public void btnNew_Click(object sender, EventArgs e)
		{
			EnableControls();
			InitControls();
		}
		
		public void ceQty_KeyPress(object sender, KeyPressEventArgs e)
		{
			if (e.KeyChar == '\r')
			{
				if (ceInner.Value >= 2)
				{
					cePcs.Value = (int) (ceInner.Value * ceQty.Value);
					btnSave.Focus();
				}
				else
				{
					MessageBox.Show("Quantity must be at least greater than or equal to 2.Make sure the item to be converted is not a PC item");
				}
				
			}
		}
		
		public void btnItemsearch2_Click(object sender, EventArgs e)
		{
			frmItemSearch frmitemserch2 = new frmItemSearch();
			frmitemserch2.ShowDialog();
			txtitem2.Text = string.Empty;
			txtstckid2.Text = string.Empty;
			txtitem2.Text = frmitemserch2.vItem.ToString();
			txtstckid2.Text = frmitemserch2.vStckid.ToString();
			if (txtstckid2.Text == string.Empty)
			{
				return;
			}
			else
			{
				ItemsearchExecute();
			}
			ceQty.Focus();
		}
		
		public void ceQty_LostFocus(object sender, EventArgs e)
		{
			
			
		}
	}
}

File: frmConvertStocks.vb
Imports PDSA.DataLayer.DataClasses
Imports PDSA.Validation
Imports DevExpress.XtraEditors
Public Class frmConvertStocks
    Private WithEvents TranBo As PDSATransaction
    Dim mgr8 As stocksManager
    Dim mOrderId As Integer = 0
    Dim Tran2 As PDSATransaction
    Dim Tran3 As PDSATransaction
    Private Sub frmConvertStocks_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        Me.Dispose()
    End Sub

    Private Sub frmConvertStocks_FormClosing(sender As Object, e As FormClosingEventArgs) Handles Me.FormClosing
        If btnSave.Enabled = True Then
            e.Cancel = True
            MessageBox.Show("Save or Cancel the Conversion of Items First Before Closing the Form.")
        End If
    End Sub

    Private Sub frmConvertStocks_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Me.Cursor = Cursors.WaitCursor


        Me.Cursor = Cursors.Default
    End Sub

    Private Sub btnItemSearch_Click(sender As Object, e As EventArgs) Handles btnItemSearch.Click
        Dim frmitemserch As New frmItemSearch
        frmitemserch.ShowDialog()
        txtItem.Text = String.Empty
        txtStckId.Text = String.Empty
        txtItem.Text = frmitemserch.vItem.ToString
        txtStckId.Text = frmitemserch.vStckid.ToString
        ceInner.Value = frmitemserch.vInnerQty
        ceAvlbl.Value = frmitemserch.vAvlbl

        If txtStckId.Text = String.Empty Then
            Exit Sub
        Else
            ItemsearchExecute()
        End If
        btnItemsearch2.Focus()
    End Sub

    Private Sub ItemsearchExecute()
        Dim col2 As stocksCollection
        Dim nStckid As Integer = 0
        Try
            mgr8 = New stocksManager()
            mgr8.DataObject.WhereFilter = stocksData.WhereFilters.PrimaryKey
            mgr8.DataObject.Entity.stckid = CInt(txtStckId.Text)
            col2 = mgr8.BuildCollection()
            If mgr8.DataObject.RowsAffected > 0 Then
                nStckid = mgr8.DataObject.Entity.stckid
                'txtPrice.Text = FormatNumber(CStr(mgr8.DataObject.Entity.cost), 2)
                ceQty.Focus()
            End If

        Catch ex As PDSAValidationException
            MessageBox.Show(ex.Message)
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
        End Try
    End Sub

    Private Sub btnClose_Click(sender As Object, e As EventArgs) Handles btnClose.Click
        Me.Close()
    End Sub

    Private Sub btnSave_Click(sender As Object, e As EventArgs) Handles btnSave.Click
        If cePcs.Value <= 1 Then
            MessageBox.Show("Cannot convert quantity")
            Exit Sub
        End If
        If ceQty.Value > ceAvlbl.Value Then
            MessageBox.Show("Quantity to be converted must not be greater than the available stock.")
            Exit Sub
        End If
        If txtItem.Text = String.Empty Then
            MessageBox.Show("Please select an item to be converted.")
            txtItem.Focus()
            Exit Sub
        End If

        If ceQty.Value <= 0 Then
            MessageBox.Show("Please enter the number of items to be converted.")
            ceQty.Focus()
            Exit Sub
        End If

        If deBoDate.Text = String.Empty Then
            MessageBox.Show("Please enter the date the item was converted.")
            deBoDate.Focus()
            Exit Sub
        End If

        If ceInner.Value < 2 Then
            MessageBox.Show("The Item Innerbox Qty is not valid and should be at least 2.")
            ceInner.Focus()
            Exit Sub
        End If


        TranBo = New PDSATransaction() ' did not change but the transaction is about converting items to pieces
        Tran2 = New PDSATransaction()
        Tran3 = New PDSATransaction()
        Dim mgrbo As convertstckManager = New convertstckManager() ' mgrbo is just a variable but it is pertaining to the convertstck table

        Try
            mgrbo.DataObject.TransactionType = PDSATransactionType.Insert
            mgrbo.DataObject.Entity.innerqty = CInt(ceInner.Value)
            mgrbo.DataObject.Entity.condate = CDate(deBoDate.Text)
            mgrbo.DataObject.Entity.fromstckid = CInt(txtStckId.Text)
            mgrbo.DataObject.Entity.tostckid = CInt(txtstckid2.Text)
            mgrbo.DataObject.Entity.fromqty = CInt(ceQty.Value)
            mgrbo.DataObject.Entity.InsertedBy = PDSAAppConfig.CurrentLoginID
            mgrbo.DataObject.Entity.ModifiedBy = PDSAAppConfig.CurrentLoginID
            mgrbo.DataObject.Entity.InsertedOn = DateTime.Now
            mgrbo.DataObject.Entity.toqty = CInt(cePcs.Value)
            TranBo.Add(mgrbo.DataObject)
            TranBo.Execute()

            Dim mgrspBo As spConvertFirstManager = New spConvertFirstManager()
            mgrspBo.Entity.StckIid = CInt(txtStckId.Text)
            mgrspBo.Entity.Convertid = mOrderId
            Tran2.Add(mgrspBo.DataObject)
            'TranBo.Add(mgrspBo.DataObject)

            Dim mgrspconsec As spConvertsecondManager = New spConvertsecondManager()
            mgrspconsec.Entity.StckIid = CInt(txtstckid2.Text)
            mgrspconsec.Entity.Convertid = mOrderId
            Tran2.Add(mgrspconsec.DataObject)
            'TranBo.Add(mgrspconsec.DataObject)

            Tran2.Execute()
            'TranBo.Execute()
            DisableControls()
            MessageBox.Show("Conversion Entry Successfully Saved", "Saved")
            InitControls()
        Catch ex As PDSAValidationException
            MessageBox.Show(ex.Message)
            Exit Sub
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
            Exit Sub
        End Try

    End Sub
    Private Sub TranBo_AfterSubmit(ByVal sender As Object, ByVal e As PDSA.DataLayer.DataClasses.PDSATransactionEventArgs) Handles TranBo.AfterSubmit
        ' Get the SQL Identity generated and store into local variable
        If e.ClassName = "convertstckData" Then
            mOrderId = DirectCast(e.DataClassTable.EntityObject, convertstck).convertid
        End If
    End Sub

    Private Sub Tran_BeforeSubmit(ByVal sender As Object, ByVal e As PDSA.DataLayer.DataClasses.PDSATransactionEventArgs) Handles TranBo.BeforeSubmit
        ' Put local order id into line item objects prior to submitting INSERT statement
        If e.ClassName = "convertstckData" Then
            DirectCast(e.DataClassTable.EntityObject, convertstck).convertid = mOrderId
        End If
    End Sub

    Private Sub EnableControls()
        btnSave.Enabled = True
        btnCancel.Enabled = True
        GroupControl1.Enabled = True
        btnClose.Enabled = False
        btnNew.Enabled = False
        deBoDate.Focus()
    End Sub

    Private Sub DisableControls()
        btnSave.Enabled = False
        btnCancel.Enabled = False
        GroupControl1.Enabled = False
        btnClose.Enabled = True
        btnNew.Enabled = True
        btnNew.Focus()
    End Sub

    Private Sub InitControls()
        deBoDate.EditValue = DateAndTime.Today
        txtItem.Text = ""
        txtitem2.Text = ""
        txtStckId.Text = ""
        txtstckid2.Text = ""
        ceQty.Value = 0
        cePcs.Value = 0
        ceAvlbl.Value = 0

    End Sub

    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        DisableControls()
        InitControls()
    End Sub

    Private Sub btnNew_Click(sender As Object, e As EventArgs) Handles btnNew.Click
        EnableControls()
        InitControls()
    End Sub

    Private Sub ceQty_KeyPress(sender As Object, e As KeyPressEventArgs) Handles ceQty.KeyPress
        If e.KeyChar = Chr(13) Then
            If ceInner.Value >= 2 Then
                cePcs.Value = CInt(ceInner.Value * ceQty.Value)
                btnSave.Focus()
            Else
                MessageBox.Show("Quantity must be at least greater than or equal to 2.Make sure the item to be converted is not a PC item")
            End If

        End If
    End Sub

    Private Sub btnItemsearch2_Click(sender As Object, e As EventArgs) Handles btnItemsearch2.Click
        Dim frmitemserch2 As New frmItemSearch
        frmitemserch2.ShowDialog()
        txtitem2.Text = String.Empty
        txtstckid2.Text = String.Empty
        txtitem2.Text = frmitemserch2.vItem.ToString
        txtstckid2.Text = frmitemserch2.vStckid.ToString
        If txtstckid2.Text = String.Empty Then
            Exit Sub
        Else
            ItemsearchExecute()
        End If
        ceQty.Focus()
    End Sub

    Private Sub ceQty_LostFocus(sender As Object, e As EventArgs) Handles ceQty.LostFocus
        

    End Sub
End Class